File: typescript/src/renderer.ts
// import { newRectangle, Rectangle } from "./shapes/rectangle.js";
import { newRectangle } from "./shapes/rectangle.js";
import { Shape } from "./shapes/shape.js";

function newRenderer(s: Shape) {
    return {
        draw() {
            const area: number = s.computeArea()
            console.log("Shape drawn\n" + "Its area is " + area)
        }
    }
}

export { newRenderer }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/Renderer.java
package edu.cmu.cs.cs214.lab02;

import edu.cmu.cs.cs214.lab02.shapes.*;

public class Renderer{
    // public Rectangle rectangle;
    
    // Renderer(Rectangle rectangle) {
    //     this.rectangle = rectangle;
    // }

    private Shape s;

    Renderer(Shape s){
        this.s = s;
    }


    void draw() {

        double area = s.getArea();

        // assume implementation

        System.out.println("Shape printed\n" + "Its area is " + area);
    }
}
